subsys/settings/src/settings.c

```diff
@@ -38,17 +38,16 @@ void settings_init(void)
 #if defined(CONFIG_SETTINGS_DYNAMIC_HANDLERS)
 int settings_register(struct settings_handler *handler)
 {
-	int rc;
-
-	k_mutex_lock(&settings_lock, K_FOREVER);
+	int rc = 0;
 
 	Z_STRUCT_SECTION_FOREACH(settings_handler_static, ch) {
 		if (strcmp(handler->name, ch->name) == 0) {
```

Comment:
Can another thread write to `handler->name` in parallel?

Reply:
Yes it is possible. Settings is responsible for protection of  its internal structures only. `handler->name` and `ch->name` may both point to area that may be RW. In case of `ch->name` modification is harder to be done, as there is no API in settings that would allow that access, so modification requires  addressing via linker symbols. In case of `handler->name`, which is provided by caller, we have no control over what happens with the variable outside of `settings `and this is `settings `user responsibility to keep the `handler ` structure integrity till the `settings_register `function returns.